src/Chapter2.hs

```diff
@@ -336,7 +338,13 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList _ _ [] = []
+subList m n (x:xs) 
+  | m < 0 || n < 0 = []
+  | m > n = []
+  | m == n = [x]
+  | m >= length (x:xs) = []
```

Comment:
Calculating the list of the list is an expensive operation. Also, according to our task, if `m` is bigger then the list length, then it's okay to return elements to the end of the list. So no need to calculate length of a list :slightly_smiling_face: